### PR TITLE
Put mod_inbox in a map with defaults

### PR DESCRIPTION
**Pull request by NelsonVides**
Wanted to come back to the topic of async workers for inbox but it would be good to have this one already, less conflicts later 🙂

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/106606) / small_tests / 07a429768761e65ecdfe3fbe994b98a7b38f3470
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106606/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106606/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-16_17.30.37/lib.mongooseim.logs/run.2022-02-16_17.30.38/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/106659) / small_tests / aeeff777298ac3003be8e7b71107484bd98de9f4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106659/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106659/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-16_17.58.41/lib.mongooseim.logs/run.2022-02-16_17.58.41/suite.log.html)

File: test/config_parser_SUITE.erl [hunk 1]
```diff
@@ -1717,23 +1717,24 @@ mod_extdisco(_Config) ->
...
 
 mod_inbox(_Config) ->
+    check_module_defaults(mod_inbox),
+    check_iqdisc_map(mod_inbox),
+    P = [modules, mod_inbox],
     T = fun(Opts) -> #{<<"modules">> => #{<<"mod_inbox">> => Opts}} end,
-    M = fun(Opts) -> modopts(mod_inbox, Opts) end,
     ChatMarkers = [<<"displayed">>, <<"received">>, <<"acknowledged">>],
-    ?cfgh(M([{reset_markers, ChatMarkers}]),
+    ?cfgh(P ++ [reset_markers, ChatMarkers],
```
Comment (chrzaszcz) [inline test/config_parser_SUITE.erl:1725]:
```suggestion
    ?cfgh(P ++ [reset_markers], ChatMarkers,
```
Same applies to the lines below, please see https://github.com/esl/MongooseIM/pull/3548 as well.

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general, I have a few minor comments. A mistake in tests allowed me to find a small bug in test helpers.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/106757) / small_tests / 2e89a8a850f7f0bf5c03dcc49e1eb3beb0ca5de6
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106757/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106757/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_09.03.16/lib.mongooseim.logs/run.2022-02-17_09.03.16/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/106791) / small_tests / 077fbe226c07cb243dce2efdf05b75e0404ee6d3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106791/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106791/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-02-17_09.19.05/lib.mongooseim.logs/run.2022-02-17_09.19.06/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/106873) / small_tests / b7547c79d483125ec980f1db30252867d898c832
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106873/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106873/small_tests.23.1/small/ct_run.mongooseim@localhost.2022-02-17_09.53.16/lib.mongooseim.logs/run.2022-02-17_09.53.17/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/106986) / small_tests / 64699013606db578de1cddbae7535e62f964f4d5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3547/106986/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3547/106986/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-02-17_10.08.35/lib.mongooseim.logs/run.2022-02-17_10.08.36/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good!

**Pull request merged**